host/integration_test.go

```diff
@@ -1019,7 +1019,7 @@ func (s *integrationSuite) TestWorkflowRetry() {
 		previousStartedEventID, startedEventID int64, history *workflow.History) ([]byte, []*workflow.Decision, error) {
 		executions = append(executions, execution)
 		attemptCount++
-		if attemptCount == 3 {
```

Comment:
why is this not catching that attempt issue?

Reply:
The issue is workflow goes into an endless retry mode.  So attempt incremented within the workflow will reach 3.  The bugfix is to bailout the workflow due to hitting max attempts for retry policy.